tests/test_plan_stubs.py

```diff
@@ -26,3 +28,47 @@ def measure_plan(detectors):
     RE(measure_plan([hw.motor, hw.noisy_det]), {'event': [cnt]})
     # Check that we saw the right number of events
     assert cnt.value == 250
+
+
+@pytest.mark.timeout(10)
+def test_monitor_events(RE, hw):
+    logger.debug('test_monitor_events')
+
+    def plan(detectors, events=None, duration=None):
+        yield from open_run()
+        yield from monitor_events(detectors, events=events, duration=duration)
+        yield from close_run()
+
+    detectors = [hw.det]
+
+    # should take about 0.3s
+    start = time.time()
+    RE(plan(detectors, duration=0.3))
+    delta = time.time() - start
+    assert 0.3 < delta < 1.3
+
+    for det in detectors:
+        det.put(0)
+
+    def bg(loop):
+        logger.debug('bg')
+        for det in detectors:
+            value = det.get()
+            det.put(value + 1)
+        logger.debug(value)
+        if value < 10:
+            loop.call_later(0.1, bg, loop)
```

Comment:
There is an `ophyd.sim` device that updates at a set frequency. Using this might simplify your test here.

https://github.com/NSLS-II/ophyd/blob/f4fdae506ac33b1611471eb7f5a406c3b39e1f0c/ophyd/sim.py#L161

Reply:
Thank you, this is much better. I'll simplify.